File: package/main/src/Date/isLeapYear.ts
/**
 * Determine if a given year is a leap year
 * @param {number} year - The year to check
 * @returns {boolean} true if the year is a leap year, false otherwise
 * @example
 * isLeapYear(2000); // Returns true (divisible by 400)
 * isLeapYear(2020); // Returns true (divisible by 4 but not 100)
 * isLeapYear(2100); // Returns false (divisible by 100 but not 400)
 * isLeapYear(2023); // Returns false (not divisible by 4)
 */
export const isLeapYear = (year: number): boolean => {
  if (!Number.isInteger(year)) {
    return false;
  }

  return (year % 4 === 0 && year % 100 !== 0) || year % 400 === 0;
};

File: package/umt_python/src/date/is_leap_year.py
def is_leap_year(year: int) -> bool:
    """
    Determine if a given year is a leap year.

    Args:
        year: The year to check

    Returns:
        True if the year is a leap year, False otherwise

    Example:
        >>> is_leap_year(2000)
        True
        >>> is_leap_year(2020)
        True
        >>> is_leap_year(2100)
        False
        >>> is_leap_year(2023)
        False
    """
    if not isinstance(year, int) or year != int(year):
        return False

    return (year % 4 == 0 and year % 100 != 0) or year % 400 == 0
